compiler/modules/print.llvm/llvm.bal

```diff
@@ -71,9 +71,53 @@ public function constInt(IntType ty, int val) returns ConstValue {
 
 // Corresponds to LLVMConstReal
 public function constReal(RealType ty, float val) returns ConstValue {
-    // XXX #307 use decimal for non-special floats 
-    return new ConstValue(ty, "0x" + int:toHexString(float:toBitsInt(val)));
-
+    string valRep;
+    if val.isInfinite() {
+        if val > 0.0 {
+            valRep = "0x7FF0000000000000";
```

Comment:
Clearer to do use a final with something like (1.0/0.0).toBitsInt().toHexString().

Reply:
Fixed